### PR TITLE
Setup Codeql workflow 

### PR DESCRIPTION
**Pull request by mmorel-35**
This workflow provides a Static Application Security Testing to comply with OSSF rules.
This will provide security recommandations in the security tab of the project along with ossf scorecard. Those are complementary and Codeql is also preconised by scorecard.

Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by alecholmez** [comment]
Hi @mmorel-35 can you please provide some extra information regarding this change? Intentions, needs, etc... what is this solving? Thanks!

**Comment by mmorel-35** [comment]
Ok, done in the pr description.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  push:
+    branches:
+    - "main"
+  pull_request:
+    branches:
+    - "main"
+  schedule:
+  - cron: '16 11 * * 5'
+
+permissions:
```
Comment (phlax) [inline .github/workflows/codeql.yml:13]: nit: im tryhing to make a ~standard of putting this first after the name - similar in jobs secrets/permissions first so they are more obvious

**Review by phlax — APPROVED** [review]
lgtm, thanks @mmorel-35

**Pull request merged**